cmd/goal/node.go

```diff
@@ -446,11 +449,11 @@ var createCmd = &cobra.Command{
 
 		// build config and save to destination
 		localConfig := config.GetDefaultLocal()
-		localConfig.Archival = newNodeArchival
+		localConfig.Archival = newNodeArchival || newNodeRelay != "" || newNodeIndexer
 		localConfig.IsIndexerActive = newNodeIndexer
 		localConfig.RunHosted = runUnderHost
-		if peerDial != "" {
-			localConfig.NetAddress = peerDial
+		if newNodeRelay != "" {
```

Comment:
Sorry - one more thing here.  We should probably validate the listenIP and newNodeRelay IP addresses.  They should be run through an ip parser to ensure they're valid IPv4 and/or IPv6 addresses and must specify a port #.  This is bonus credit though, so if it looks too complicated we can punt for now.  If we were Microsoft and this was going to 1e7 people, I would be more careful.